Content.Shared/SimpleStation14/Species/Shadowkin/Systems/ShadowkinPowerSystem.cs

```diff
@@ -0,0 +1,282 @@
+            var result = ShadowkinPowerThreshold.Min;
+            var value = ShadowkinComponent.PowerThresholds[ShadowkinPowerThreshold.Max];
+
+            // Find the highest threshold that is lower than the current power level
+            foreach (var threshold in ShadowkinComponent.PowerThresholds)
+            {
+                if (threshold.Value <= value && threshold.Value >= powerLevel)
+                {
+                    result = threshold.Key;
+                    value = threshold.Value;
+                }
+            }
+
+            var powerDictionary = new Dictionary<ShadowkinPowerThreshold, string>
+            {
+                {ShadowkinPowerThreshold.Max, Loc.GetString("shadowkin-power-max")},
+                {ShadowkinPowerThreshold.Great, Loc.GetString("shadowkin-power-great")},
+                {ShadowkinPowerThreshold.Good, Loc.GetString("shadowkin-power-good")},
+                {ShadowkinPowerThreshold.Okay, Loc.GetString("shadowkin-power-okay")},
+                {ShadowkinPowerThreshold.Tired, Loc.GetString("shadowkin-power-tired")},
+                {ShadowkinPowerThreshold.Min, Loc.GetString("shadowkin-power-min")}
+            };
+
+            // Return the name of the threshold
+            powerDictionary.TryGetValue(result, out var powerType);
+            return powerType ?? Loc.GetString("shadowkin-power-okay");
+        }
+
+        /// <summary>
+        ///    Sets the alert level of a shadowkin.
+        /// </summary>
+        /// <param name="uid">The entity uid.</param>
+        /// <param name="enabled">Enable the alert or not</param>
+        /// <param name="powerLevel">The current power level.</param>
+        public void UpdateAlert(EntityUid uid, bool enabled, float? powerLevel = null)
+        {
+            if (!enabled || powerLevel == null)
+            {
+                _alertsSystem.ClearAlert(uid, AlertType.ShadowkinPower);
+                return;
+            }
+
+            // Get shadowkin component
+            if (!_entityManager.TryGetComponent<ShadowkinComponent>(uid, out var component))
+            {
+                Logger.Error("Tried to update alert of entity without shadowkin component.");
+                throw new InvalidOperationException("Tried to update alert of entity without shadowkin component.");
+            }
+
+            // Get the power as a short from 0-5
+            var power = ContentHelpers.RoundToLevels((double) powerLevel, component.PowerLevelMax, 8);
+
+            // Set the alert level
+            _alertsSystem.ShowAlert(uid, AlertType.ShadowkinPower, (short) power);
+        }
+
+
+        /// <remarks> For viewing purposes. </remarks>
+        /// <param name="powerLevel">The current power level.</param>
+        /// <returns>Power level as an integer.</returns>
+        public int GetLevelInt(float powerLevel)
+        {
+            // Very dumb, round and convert to int
+            return (int) Math.Round(powerLevel);
+        }
```

Comment:
Does this _really_ need to be a function? Couldn't you just use (int) Math.Round(powerLevel); ?

Reply:
it doesn't :)